### ISSUE TITLE
Illegal type annotation: variable not allowed unless it is a type alias

### ISSUE THREAD

**Issue by tekumara** [microsoft/pyright#4063]
```python
from datetime import date

class LineItem:
    date: date
    created_date: date
```

```
pyright 1.1.275
/Users/tekumara/code/test.py
  /Users/tekumara/code/test.py:5:19 - error: Illegal type annotation: variable not allowed unless it is a type alias (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
Completed in 0.455sec
```

If I rename `LineItem.date` to something else the error goes away.

**Comment by erictraut** [comment]
Here's a typical workaround for a namespace collision like this:

```python
from datetime import date as _date

class LineItem:
    date: _date
    created_date: _date
```
